WiseBuy/Managers/DatabaseManager.m

```diff
@@ -243,13 +243,13 @@ + (PFFileObject *)getPFFileFromImageData: (NSData *)imageData {
     return [PFFileObject fileObjectWithName:@"image.png" data:imageData];
 }
 
-+ (void)fetchDeals:(PFObject *)item withCompletion:(void(^)(NSArray *deals ,NSError *error))completion {
++ (void)fetchDeals:(PFObject *)item withCompletion:(void(^)(NSArray<Deal *> *deals ,NSError *error))completion {
     
     PFQuery *dealsQuery = [Deal query];
     [dealsQuery includeKey:kItem];
     [dealsQuery whereKey:kItem equalTo:item];
     
-    [dealsQuery findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+    [dealsQuery findObjectsInBackgroundWithBlock:^(NSArray<Deal *> * _Nullable objects, NSError * _Nullable error) {
```

Comment:
ah you're right that I could specify PFQuery with generics too. 
If I change the `NSArray<Deal *>` to `NSArray<Item *>, it does generate a warning for that

Reply:
Oh wait, so changing it to `NSArray<Item *>` generates a compile-time warning? I didn't expect that.

The general idea here is that when the array is initially created, the compiler should be told the type. So for DatabaseManager where you're creating the array, you can specify the correct generic type. For this Parse one, you don't create the array--it's passed through findObjectsInBackgroundWithBlock, so we need to check that API. The doc says it can return a generic array (https://parseplatform.org/Parse-SDK-iOS-OSX/api/Type%20Definitions.html#/c:PFQuery.h@T@PFQueryArrayResultBlock), but I assumed for it to know the right type, you'd need to pass it to `PFQuery`. If the compiler is able to warn you about the mismatch though, then it seems like it's doing the right thing already.

Reply:
I saw the warning just by changing the array to Item <without needing to update PFQuery>
I think this works without updating the PFQuery because when defining the method, I already `NSArray<Deal *> *deals`